samples/basic/blinky/src/main.c

```diff
@@ -37,6 +39,9 @@ int main(void)
 		if (ret < 0) {
 			return 0;
 		}
+
+		led_state = !led_state;
+		printf("LED state: %s\n", led_state ? "ON" : "OFF");
```

Comment:
The whole point of this sample is you can just visually see without needing a UART (which is actually great when porting an architecture/board because you don't need a working UART driver), this throws a massive spanner in the works for that type of testing/usage going forward

Reply:
I understand what you are saying, yes the binary is going to link against stdio (the `printf` symbol will be present in the binary), but if the board has no working UART (implies `CONFIG_CONSOLE`, `CONFIG_SERIAL` disabled), `char_out` will never be called.

Please note that after applying those changes even if a board has no UART devices/drivers the sample will continue the work just as before (i.e. the LED will blink).
I've tested this by setting the above mentioned configs to `n`, and removing UART devices from the device tree. The console output part is completely optional, but if enabled provides a valuable feedback. 


Moreover, the `blinky_pwm` sample already prints output to the console in the case of timer errors.

Reply:
I believe that having an additional feedback on UART is a nice bonus - it doesn't change the original purpose of the sample, but might help with debugging problems.

Reply:
To see why this is not good, build this sample with https://github.com/zephyrproject-rtos/zephyr/pull/63333 for a board with a USB CDC output (or hardware flow control) with LOG enabled then watch what happens - on an nrf52840dongle, it blinks for a bit then the LED stays on permanently, until someone opens the UART and empties the buffer. This then stops being blinky and is more constant light/lack of light.

Reply:
I've been following the conversation in #63333, and as mentioned there, I
think that this should be at least configurable, and preferably opt-in.

There are a lot of applications where UART is used only as an additional
debug/supplementary tool (such as this change). In such cases blocking by 
default could be confusing for the users, as dropping packets historically did
not block (unless hardware flow control was implemented).

Reply:
Since the #63333 has been merged, and the changes related to blocking behavior for CDC ACM UART are opt-in (`hw-flow-control` flag), this PR doesn't negatively influence the sample behavior @nordicjm